### PR TITLE
updated uni-algo to support building the library

### PR DESCRIPTION
**Pull request by bigerl**
Specify library name and version:  **uni-algo/0.7.1**

This PR does two things: bump the version of uni-algo to 0.7.1 and extends it to a built library. 

The existing recipe provides uni-algo as a header-only library. This comes with limitation on the side of uni-algo. Therefore, I've extended the existing recipe by an option `header_only` that allows to control if it should be header only or built. 

Open question:
- Should the recipe for the existing versions be changed? If no, is there any example how to manage different recipes for different version?

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/uni-algo/all/conanfile.py [hunk 1]
```diff
@@ -1,66 +1,146 @@
...
 from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
 import os
 
-required_conan_version = ">=1.52.0"
+required_conan_version = ">=1.53.0"
+
 
-class UniAlgoConan(ConanFile):
+class PackageConan(ConanFile):
```
Comment (StellaSmith) [inline recipes/uni-algo/all/conanfile.py:14]:
```suggestion
class UniAlgoConan(ConanFile):
```

File: recipes/uni-algo/all/conanfile.py [hunk 2]
```diff
@@ -1,66 +1,146 @@
...
     name = "uni-algo"
```
Comment (StellaSmith) [inline recipes/uni-algo/all/conanfile.py:15]:
```suggestion
    name = "uni-algo"
    package_type = "library"
```

**Review by StellaSmith — CHANGES_REQUESTED** [review]

**Review by StellaSmith — DISMISSED** [review]

File: recipes/uni-algo/all/conanfile.py [hunk 3]
```diff
@@ -1,66 +1,147 @@
...
+                    self.cpp_info.libs = [f"{self.name}-debug"]
+            else:
+                self.cpp_info.libs = [f"{self.name}"]
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                self.cpp_info.system_libs.append("m")
+                self.cpp_info.system_libs.append("pthread")
+                self.cpp_info.system_libs.append("dl")
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        self.cpp_info.filenames["cmake_find_package"] = f"{self.name}".upper()
```
Comment (prince-chrismc) [inline recipes/uni-algo/all/conanfile.py:144]: There's look like extras from the template, does upstream install cmake config files?

Comment (bigerl) [reply recipes/uni-algo/all/conanfile.py:144]: I guess you mean the upstream of uni-algo. It actually does but only if the CMake option [UNI_ALGO_INSTALL is set](https://github.com/uni-algo/uni-algo/blob/main/CMakeLists.txt#L15). 

How would I handle this best? I would suggest set that option unconditionally in the `generate` method like:
```python
tc.variables["UNI_ALGO_INSTALL"] = True
```

Do you agree?

Comment (prince-chrismc) [reply recipes/uni-algo/all/conanfile.py:144]: If you add a link to the CMake + use the Conan  2 way for the new generators with `set_property` (we have some docs in the v2 migration page) 

That would be superb 👌

Comment (bigerl) [reply recipes/uni-algo/all/conanfile.py]: @prince-chrismc I've patched. hope I did things right. CMake and how it is modeled in conan has still some mysterious corners for me. 

The CI failed on the commit without any information that helped me to figure out why. If you have access to the backend could you check what's wrong there?

In an blind attempt to fix it I merged master. Let's see if that helps something.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**